engine/src/w32dc.cpp

```diff
@@ -299,16 +299,30 @@ void *MCScreenDC::GetNativeWindowHandle(Window p_win)
 // IM-2014-01-28: [[ HiDPI ]] Return the x & y dpi of the main screen
 bool MCWin32GetScreenDPI(uint32_t &r_xdpi, uint32_t &r_ydpi)
 {
-	HDC t_dc;
-	t_dc = GetDC(NULL);
+	static bool s_fetched = false;
+	static uint32_t s_xdpi, s_ydpi;
+	if (!s_fetched)
+	{
+		s_fetched = true;
 
-	if (t_dc == NULL)
-		return false;
+		HDC t_dc;
+		t_dc = GetDC(NULL);
+
+		if (t_dc == NULL)
+		{
+			s_xdpi = 96;
+			s_ydpi = 96;
+			return false;
+		}
 
-	r_xdpi = GetDeviceCaps(t_dc, LOGPIXELSX);
-	r_ydpi = GetDeviceCaps(t_dc, LOGPIXELSY);
+		s_xdpi = s_xdpi = GetDeviceCaps(t_dc, LOGPIXELSX);
```

Comment:
?? a double assignment?

Reply:
Ooops - it was a copy paste error as I tweaked my original attempt. This code is going to change anyway - it would be better to cache the metrics in the MCScreenDC class as that is where the 'settingchange' message is handled so it can then flush the settings and refetch them.